drgn/helpers/linux/bpf.py

```diff
@@ -180,3 +183,59 @@ def cgroup_bpf_prog_for_each_effective(
             if not prog:
                 break
             yield prog
+
+
+def bpf_prog_used_maps(bpf_prog: Object) -> Iterator[Object]:
+    """Yield maps used by a BPF program."""
+    for i in range(bpf_prog.aux.used_map_cnt.value_()):
+        yield bpf_prog.aux.used_maps[i]
```

Comment:
This requires reading `bpf_prog.aux` twice, so as a small optimization, you can read it ahead of time. We can also use a slicing trick to optimize this further:


```suggestion
    aux = bpf_prog.aux.read_()
    return iter(aux.used_maps[:aux.used_map_cnt])
```